src/Raven.Client/Extensions/ExpressionExtensions.cs

```diff
@@ -98,16 +98,21 @@ public static string ToPropertyPath(this Expression expression, char propertySep
                 if (curValue.Length == 1 && curValue[0] == propertySeparator && i != stackLength - 1)
                 {
                     var nextVal = propertyPathExpressionVisitor.Results.Peek();
-
                     if (nextVal.Length == collectionSeparator.Length && nextVal.Equals(collectionSeparator, StringComparison.OrdinalIgnoreCase))
                     {
                         continue;
                     }
                 }
 
-                builder.Append(curValue);
+                if (curValue.Length == 6 && curValue == "$Value" && i != stackLength - 1)
```

Comment:
`curValue.Length == 6` is redundant, no? Already checked by the string comparer.

Reply:
done